apple/job/__main__.py

```diff
@@ -61,6 +64,24 @@
 
 date_range = get_date_range(START_DATE, END_DATE)
 
+# check if all needed environment variables are set
+critical_env_var_missing = False
+if not APPLE_AUTOMATION_ENDPOINT:
+    logger.error("APPLE_AUTOMATION_ENDPOINT is not set")
+    critical_env_var_missing = True
+if not APPLE_AUTOMATION_BEARER_TOKEN:
+    logger.error("APPLE_AUTOMATION_BEARER_TOKEN is not set")
+    critical_env_var_missing = True
+if not APPLE_PODCAST_ID:
+    logger.error("APPLE_PODCAST_ID is not set")
+    critical_env_var_missing = True
+if not OPENPODCAST_API_TOKEN:
+    logger.error("OPENPODCAST_API_TOKEN is not set")
+    critical_env_var_missing = True
+if critical_env_var_missing == True:
+    logger.error("Critical environment variables are missing, exiting")
+    exit(1)
+
```

Comment:
Alternatively (which is a bit more idiomatic I'd say):

```suggestion
required_env_vars = [
    "APPLE_AUTOMATION_ENDPOINT",
    "APPLE_AUTOMATION_BEARER_TOKEN",
    "APPLE_PODCAST_ID",
    "OPENPODCAST_API_TOKEN"
]

for env_var in required_env_vars:
    if not os.getenv(env_var):
        logger.error(f"{env_var} is not set")
        exit(1)
```

Reply:
but it doesn't do the same ;) but I know what you mean...

Reply:
Nevermind. It's not the same. 😅 All good then.

Reply:
better now? https://github.com/openpodcast/pipelines/blob/e7e28324d942ab03e0b08e611805272b1a5fa1bd/apple/job/__main__.py#L68